File: dz-playlist/src/main/java/com/queryinterface/dz/playlist/kafka/messages/TrackSearchCommand.java
package com.queryinterface.dz.playlist.kafka.messages;

import com.queryinterface.dz.playlist.Track;

public record TrackSearchCommand(long playlistId,
                              long trackId,
                              String title,
                              String artist) implements KafkaMessage {
    
    public static TrackSearchCommand of(Track track) {
        return new TrackSearchCommand(track.getPlaylistId(), track.getId(), track.getName(), track.getArtist());
    }

    @Override
    public String getType() {
        return "track.search.v1";
    }
}

File: yt-scraper/Kafka/TrackSearchCommand.cs
namespace yt_scraper
{
    public class TrackSearchCommand : Message
    {
        
        public long PlaylistId { get; set; }
        public long TrackId { get; set; }
        public string Title { get; set; }
        public string Artist { get; set; }

        public string GetMessageType()
        {
            return "track.search.v1";
        }
    }
}